ansible/roles/test/tasks/advanced-reboot.yml

```diff
@@ -124,14 +138,18 @@
             src: "{{ new_image_location }}"
             dest: "{{ new_image_location }}"
 
+        - name: Schedule to restore current image
+          shell: show version
```

Comment:
>show version [](start = 17, length = 12)

This task is not necessary.

Reply:
That is for executing notify. Notify has to come with a task.

Reply:
Can you accomplish this without `notify`?

Reply:
That was what I tried first. Unfortunately notify has to be used with a task. We end up with running a dummy local task.

Reply:
Understood. My suggestion was to avoid using `notify` altogether. Can you not create a block with two tasks and make it conditional?

Reply:
I also tried debug. that didn't trigger notify.

Reply:
What is the reason you feel you must use `notify`?

Reply:
@jleveque I think you suggestion makes sense in a way.

The original code was there so that it would leverage some standard operations. For instance, you don't want to just reboot and move on. Reboot requires certain wait time for the system to stabilize. If you don't do wait, the next few tests might fail.

These notify actions captures such knowledge so that we could use in the tests.

But ultimately, let's just move to pytest :-)

Reply:
I finally remembered why we did the notify/handler thing: in the 'always' block, if anything fails in the middle of the list, the 'always' block will be cut short.

These 2 tasks were meant to be run at the very end of the sequence so that it doesn't interfere with other error recovery and/or log collection tasks in the 'always' blocks.